src/drt/src/dr/FlexDR_init.cpp

```diff
@@ -1748,8 +1745,15 @@ void FlexDRWorker::initTrackCoords(
 void FlexDRWorker::initGridGraph(const frDesign* design)
 {
   // get all track coords based on existing objs and aps
-  std::map<frCoord, std::map<frLayerNum, frTrackPattern*>> xMap;
-  std::map<frCoord, std::map<frLayerNum, frTrackPattern*>> yMap;
+  boost::container::
+      flat_map<frLayerNum, boost::container::flat_map<frCoord, frTrackPattern*>>
+          xMap;
+  boost::container::
+      flat_map<frLayerNum, boost::container::flat_map<frCoord, frTrackPattern*>>
+          yMap;
+  size_t layerCount = design->getTech()->getLayers().size();
+  xMap.reserve(layerCount);
+  yMap.reserve(layerCount);
```

Comment:
```suggestion
  xMap.reserve(layerCount + 1);
  yMap.reserve(layerCount + 1);
```
To account for xMap[-10] and yMap[-10]